analyses/org.eclipse.tracecompass.incubator.executioncomparision.core/src/org/eclipse/tracecompass/incubator/internal/executioncomparison/core/DifferentialSeqCallGraphAnalysis.java

```diff
@@ -0,0 +1,409 @@
+     * @param traceList
+     *            the list of trace names
+     * @return the list of call graphs
+     */
+    private static List<CallGraph> addToCallGraph(ITmfTimestamp start, ITmfTimestamp end, List<String> traceList) {
+        List<CallGraph> cGList = new ArrayList<>();
+        for (String traceName : traceList) {
+            ICallGraphProvider2 instrumentedCallStackAnalysis = fTraceCallGraphRegistry.get(traceName);
+            if (instrumentedCallStackAnalysis != null) {
+                ITmfTrace trace = getTrace(traceName);
+                if (trace != null) {
+                    ITmfTimestamp traceStart = start;
+                    ITmfTimestamp traceEnd = end;
+
+                    if (traceStart.getValue() < trace.getStartTime().getValue()) {
+                        traceStart = trace.getStartTime();
+                    }
+                    if (traceEnd.getValue() > trace.getEndTime().getValue()) {
+                        traceEnd = trace.getEndTime();
+                    }
+                    cGList.add(instrumentedCallStackAnalysis.getCallGraph(traceStart, traceEnd));
+                }
+            }
+        }
+        return cGList;
+    }
+
+    /**
+     * Gets the differential call graph
+     *
+     * @return DifferentialCallGraph
+     */
+    public IWeightedTreeSet<ICallStackSymbol, Object, DifferentialWeightedTree<ICallStackSymbol>> getCallGraph() {
+        DifferentialCallGraphProvider differentialCallGraphProvider = getDifferentialCallGraphProvider();
+        if (differentialCallGraphProvider != null) {
+            return differentialCallGraphProvider.getTreeSet();
+        }
+        return new DifferentialWeightedTreeSet<>();
+    }
+
+    /**
+     * Gets the differential weighted tree provider
+     *
+     * @param monitor
+     *            the progress monitor, can be null
+     * @return the differential weighted provider or null
+     */
+    public @Nullable DifferentialWeightedTreeProvider<?> getDiffProvider(@Nullable IProgressMonitor monitor) {
+        if (fTraceCallGraphRegistry.isEmpty()) {
+            InstrumentedCallStackAnalysis callGraphModule;
+            ITmfTrace trace = TmfTraceManager.getInstance().getActiveTrace();
+            Collection<ITmfTrace> traceSet = TmfTraceManager.getTraceSet(trace);
+            for (ITmfTrace traceMember : traceSet) {
+                Iterable<InstrumentedCallStackAnalysis> modules = TmfTraceUtils.getAnalysisModulesOfClass(traceMember, InstrumentedCallStackAnalysis.class);
+                for (InstrumentedCallStackAnalysis module : modules) {
+                    if (module.getId().equals(fcallStackAnalysisMap.get(traceMember.getTraceTypeId()))) {
+                        callGraphModule = module;
+                        callGraphModule.schedule();
+                        fTraceCallGraphRegistry.put(String.valueOf(traceMember.getName()), callGraphModule);
+                        callGraphModule.waitForCompletion();
```

Comment:
This loop should look at monitor and exits and cancel any jobs if the monitor is cancelled.